threatmodels/notation-theatmodel.md

```diff
@@ -0,0 +1,84 @@
+  - Artifact in registries: Take container images as an example, users can use `docker` commands to produce container images, and store them to a registry. A container image in a registry can be referenced by a fully qualified path, for example `registry.wabbit-networks.io/net-monitor@sha256:xxx`. SBOM is another example, which are generated by tools during build process, stored as OCI artifacts in OCI-compliant registries, and referenced by a fully qualified path as well as container images.
+  - Artifact as OCI image layout: OCI image layout is a directory structure that contains files and folders that refer to an OCI image. See [OCI image layout spec](https://github.com/opencontainers/image-spec/blob/v1.0.0/image-layout.md) for details.
+- **File System**: The file system where Notation or application built with Notation library is installed. For Notation signing artifacts as OCI image layout, it is the place where artifacts and associated signatures are stored. There are different directories and files as follows:
+  - Notation config file, where general configuration is stored
+  - Notation trust policy file, where trust policy configuration is stored
+  - Notation trust store, where certificate files are stored to validate signatures
+  - Notation plugin directory, where Notation plugins are installed
+  - Credential store, where the registry or Key management credentials are stored
+- **Notation Plugin**: An external component built by different vendors, and integrated to Notation as a signing plugin for signature generation, or as a verifying plugin for signature verification, see [plugin spec](https://github.com/notaryproject/notaryproject/blob/v1.0.0-rc.2/specs/plugin-extensibility.md) for details.
+- **Registry**: A registry for Docker and Open Container Initiative (OCI) images, with support for all OCI artifacts. It is the storage where artifacts are pushed to by artifact builder. Registries are outside Notation trust boundary.
+- **KMS**: A remote Key Management Service provides key and associated certificate management, and remote signing service. KMSs are outside Notation trust boundary.
+- **OCSP Responder**: The [Online Certificate Status Protocol (OCSP)](https://en.wikipedia.org/wiki/Online_Certificate_Status_Protocol) is an Internet protocol used for obtaining the revocation status of an X.509 digital certificate. Notation uses OCSP to validate certificate revocation status from OCSP Responder.
+
+## Notation sign artifacts using remote key
+
+### Data flow
+
+A signer uses Notation to sign artifacts with the private key stored in a KMS, and store signatures in the file system or in a registry.
+
+The following diagram illustrates Notation signing artifacts stored in registries with keys stored in KMS:
+
+![image](../media/notation-sign-remote.svg)
```

Comment:
"Step 3 login" is not a valid step as `login` is a pure client-side concept. Recommend to remove it.

Reply:
If you do want to focus on `login`. It will be
1. Interact
2. **Ping registry**
3. **Write registry credentials** to credential store
4. Read registry credentials
5. Fetch descriptor of the artifact
6. ...

where steps 2 & 3 together are called **Login**.